### PR TITLE
Fix coloring on cygwin/msys2 shell

### LINKED ISSUE #57: it does't working on window7(64bit)
I get wrong when run pure-go on windows.

package main  

import("github.com/fatih/color")

func main() {
      color.Red("we are red")
}

1: window7 64bit
2: git bash

output:
image

### PR DESCRIPTION
**Pull request by fatih**
Closes #57

### CODE REVIEW

File: color.go [hunk 1]
```diff
@@ -18,8 +18,7 @@ var (
 	// or not. This is a global option and affects all colors. For more control
 	// over each color block use the methods DisableColor() individually.
 	NoColor = os.Getenv("TERM") == "dumb" ||
-		!isatty.IsTerminal(os.Stdout.Fd()) ||
-		!isatty.IsCygwinTerminal(os.Stdout.Fd())
+		(!isatty.IsTerminal(os.Stdout.Fd()) || !isatty.IsCygwinTerminal(os.Stdout.Fd()))
```
Comment (k-takata) [inline color.go]: Still wrong, I think.
If both IsTerminal() and IsCygwinTerminal() return false, color cannot be used.
At least one of them returns true, color can be used.

Comment (fatih) [inline color.go]: Previously it was like:

```go
	NoColor = os.Getenv("TERM") == "dumb" ||
		!isatty.IsTerminal(os.Stdout.Fd()) ||
		!isatty.IsCygwinTerminal(os.Stdout.Fd())
```

Do you say it should be like this?

```go
	NoColor = os.Getenv("TERM") == "dumb" ||
		!isatty.IsTerminal(os.Stdout.Fd()) ||
		isatty.IsCygwinTerminal(os.Stdout.Fd())
```

I'm really confused right now. I don't have a way to test it, so can you test and give me the correct answer maybe?

**Comment by fatih** [comment]
@mattn @k-takata pushed again, ptal

**Comment by mattn** [comment]
LGTM

**Comment by fatih** [comment]
Thanks @mattn , rebased and merging once it's green

File: color.go (hunk 1 above)
Comment (k-takata) [inline color.go]: I meant:

```go
	NoColor = os.Getenv("TERM") == "dumb" ||
		!(isatty.IsTerminal(os.Stdout.Fd()) ||
		    isatty.IsCygwinTerminal(os.Stdout.Fd()))
```

This is equivalent with @mattn's suggestion.

**Comment by mattn** [comment]
```go
color.Green("YAY")
```

**Pull request merged**